flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -123,6 +130,32 @@ private FailureHandlingResult handleFailure(
                     globalFailure);
         }
 
+        if (!globalFailure) {
+            try {
+                final SchedulingExecutionVertex failedVertex =
+                        schedulingTopology.getVertex(failingExecutionVertexId);
+                if (cause instanceof PartitionException
```

Comment:
This is not correct because `PartitionException` can also happen when reading from a non-cached result.

Reply:
Maybe we can just check whether the `ResultPartitionID` of the `PartitionException` is a cached result partition. And translate it to a `CacheIntermediateDataSetCorruptedException`. This should placed above `isUnrecoverableError(...)` check.